File: models/ProductModel.js
const mongoose = require("mongoose");

const productSchema = new mongoose.Schema({
  Product: {
    type: String,
    required: true,
  
  },
  Dairy_Preference: {
    type: String,
    required: true,
  },
  rating: {
    type: Number,
  },
  Cupping_Score:{
    type: Number,
    required: true,
  },
  Sensory:{
    type: String,
    required: true,
  },
  Roast:{
    type: String,
    required: true,
  },
  Brew_Method:{
    type: String,
    required: true,
  },
  price:{
    type: Number,
    required: true,
  },
  Image:{
    type: String,
    required: true,
  },
  Quantity:{
    type: String,
    required: true,
    enum: ["250g", "500g", "750g", "1kg"]

  },
  Product_Info:{
    type: String,
    required: true,
  },
  
});

productSchema.index({ product: "text" });

const Product = mongoose.model("Product", productSchema);

module.exports = Product;

File: flutter_app/lib/features/product/data/models/product_model.dart
import 'package:skia_coffee/features/home/business/entities/product_entity.dart';

class ProductModel extends ProductEntity {
  const ProductModel({
    final String? id,
    final String? product,
    final int? cuppingScore,
    final String? roast,
    final String? brewMethod,
    final double? price,
    final String? image,
    final String? quantity,
    final String? productInfo,
    final int? v,
    final String? sensory,
  }) : super(
          id: id,
          product: product,
          roast: roast,
          brewMethod: brewMethod,
          price: price,
          quantity: quantity,
          productInfo: productInfo,
          cuppingScore: cuppingScore,
          image: image,
          v: v,
          sensory: sensory,
        );

  factory ProductModel.fromJson(Map<String, dynamic> json) {
    return ProductModel(
      id: json['_id'],
      product: json['Product'],
      cuppingScore: json['Cupping_Score'],
      sensory: json['Sensory'],
      roast: json['Roast'],
      brewMethod: json['Brew_Method'],
      price: json['price'].toDouble(),
      image: json['Image'],
      quantity: json['Quantity'],
      productInfo: json['Product_Info'],
      v: json['__v'],
    );
  }
}
